pkg/specgen/generate/namespaces.go

```diff
@@ -175,6 +175,10 @@ func namespaceOptions(ctx context.Context, s *specgen.SpecGenerator, rt *libpod.
 		if pod == nil || infraCtr == nil {
 			return nil, errNoInfra
 		}
+		// Inherit the user from the infra container if it is set
+		if infraCtr.User() != "" {
```

Comment:
@giuseppe @mheon @rhatdan does it make sense to allow containers to set `--user` when being created with `--pod`? For the `--userns=keep-id` case, the container wasn't being created with the calling user even though the uid_map was correctly set in the container, so I had to explicitly inherit the user as well from the infra container.
I am not sure if this is the right approach, but if it is, should we be able to set `--user` when creating a container in a pod with userns set?

Before explicitly setting the user to the infra user
```
podman exec -it 5be9d20f7cb14aad964fb7e8f6cc81acd3a60b8822d19631df7496f0e1b83ab7 sh
sh-5.1# id
uid=0(root) gid=0(root) groups=0(root)
sh-5.1# cat /proc/self/uid_map 
         0          1       1000
      1000          0          1
      1001       1001      64536
```

After setting it (which is what we should be getting I believe)
```
✗ ./bin/podman run --pod foo -it fedora sh
sh-5.1$ id
uid=1000(1000) gid=1000(1000) groups=1000(1000)
sh-5.1$ cat /proc/self/uid_map 
         0          1       1000
      1000          0          1
      1001       1001      64536
```

Reply:
Yes, we definitely want to allow `--user` and `--pod` to be used together; having a DB server running as user `mysql` and a web server running as user `httpd` in the same pod sounds like a very common use-case.

Reply:
Okay fixed it, here is an example output of the options
```
➜  podman git:(pod-userns) ✗ ./bin/podman pod create --userns keep-id --name foo
ae9120da729b2cf56b129fe5641f362ae06dcac6b8f3c9294f947e5f93254a13
➜  podman git:(pod-userns) ✗ ./bin/podman run --pod foo -it fedora sh           
sh-5.1$ id
uid=1000(1000) gid=1000(1000) groups=1000(1000)
sh-5.1$ 
exit
➜  podman git:(pod-userns) ✗ ./bin/podman run --pod foo -it --user 2000 fedora sh
sh-5.1$ id
uid=2000(2000) gid=0(root) groups=0(root)
sh-5.1$ 
```

Reply:
LGTM